src/main/thrift/parquet.thrift

```diff
@@ -1084,6 +1290,9 @@ struct ColumnIndex {
     * Same as repetition_level_histograms except for definitions levels.
     **/
    7: optional list<i64> definition_level_histograms;
+
+   /** A list containing statistics of GEOMETRY logical type for each page */
+   8: optional list<GeometryStatistics> geometry_stats;
```

Comment:
Why are there stats for each page? Each bbox is up to 64 bytes, which seems like a lot of overhead at the page level, especially given that WKB objects are also considerably larger than most values stored in a Parquet page.

Reply:
All fields of `GeometryStatistics` are optional and the `geometry_stats` field itself is optional. Isn't it better to provide freedom to writer implementation to turn on features they need?

Reply:
I think that it is better to give guidance and keep the spec small. I think we should only add things that have a clear use right now.

Reply:
Page level stats are super useful in a needle-in-the-haystack search. Computation on geometry type can be very slow due to its mathematical complexity. Page-level stats such as bounding box can help filtering out unnecessary pages because computation on bounding box is faster in order of magnitude than on complex polygons.

Reply:
Just to unify the conversation.  A page typically holds O(10000s of records) or O(Multiple KBs to MBs of data).  It sounds like the at least some cases it might be worth it.  I think to @rdblue point, we might want to evaluate additional file size this data adds to make sure estimates are accurate for some geography data and if it is not worth writing by default we should consider removing it.

Reply:
> we might want to evaluate additional file size this data adds to make sure estimates are accurate for some geography data and if it is not worth writing by default we should consider removing it.

@jiayuasu @Kontinuation Do you have any good benchmark data to create some testing Parquet files so we can determine if it is worth adding the column index?

Or we can remove column index of geometry type for now and add them back in the follow up to expedite the initial version.

Reply:
I think this should be removed for now. We can always add it later.

Reply:
> Do you have any good benchmark data to create some testing Parquet files so we can determine if it is worth adding the column index?

I have experimented using several datasets downloaded from [UCR STAR](https://star.cs.ucr.edu/), including NYC Taxi, PortoTaxi, Tiger LINES and MS Buildings dataset. The size of parquet data pages is about 1MB. The extra storage overhead of storing geometry stats in column index is 0.004% ~ 0.01%, which is quite close to the theoretical 64 bytes per page overhead.

The storage overhead for the column index itself is significant. Adding the geometry statistics field makes the column index 91% larger.